### PR TITLE
TEBA-65: Refund API

### PR DESCRIPTION
**Pull request by NghiaDTr**

### CODE REVIEW

File: processor/src/client/easycredit.client.ts [hunk 1]
```diff
@@ -122,6 +130,34 @@ class EasyCreditApiClient implements EasyCreditClient {
...
+      const headers: HeadersInit = { ...this.getDefaultHeaders(), ...customHeaders };
+
+      const response = await fetch(`${this.partnerBaseApiUrl}/merchant/v3/transaction/${transactionId}/refund`, {
+        method: 'POST',
+        headers,
+        body: JSON.stringify(payload),
+      });
+
+      if (response.status !== 202) {
+        log.error('Refund request returned invalid status code', response);
```
Comment (Tung-Huynh-Shopmacher) [inline processor/src/client/easycredit.client.ts:149]: throw Exception to catch below

File: processor/src/routes/payment.route.ts [hunk 2]
```diff
@@ -159,4 +166,30 @@ export const paymentsRoute = async (fastify: FastifyInstance, opts: FastifyPlugi
...
   );
+
+  fastify.post<{
+    Params: { paymentId: string };
+    Body: RefundPaymentRequestSchemaDTO;
+    Reply: RefundPaymentResponseSchemaDTO;
+  }>(
+    '/:paymentId/refund',
+    {
+      preHandler: [opts.sessionHeaderAuthHook.authenticate()],
```
Comment (Tung-Huynh-Shopmacher) [inline processor/src/routes/payment.route.ts:177]: use oauth2AuthHook

File: processor/src/validators/payment.validators.ts [hunk 3]
```diff
@@ -157,3 +162,31 @@ export const validateTransaction = (payment: Payment) => {
...
+
+  return transaction;
+};
+
+export const validatePaymentAmount = (payment: Payment, refundAmount: number): void => {
+  const { centAmount, fractionDigits } = payment.amountPlanned;
+
+  const amountInEur = convertCentsToEur(centAmount, fractionDigits);
+
+  if (refundAmount > amountInEur) {
```
Comment (Tung-Huynh-Shopmacher) [inline processor/src/validators/payment.validators.ts:185]: amount > 0

**Review by Tung-Huynh-Shopmacher — CHANGES_REQUESTED** [review]

File: processor/src/client/easycredit.client.ts (hunk 1 above)
Comment (NghiaDTr) [reply processor/src/client/easycredit.client.ts]: I fixed this

File: processor/src/routes/payment.route.ts (hunk 2 above)
Comment (NghiaDTr) [reply processor/src/routes/payment.route.ts]: I fixed this

File: processor/src/validators/payment.validators.ts (hunk 3 above)
Comment (NghiaDTr) [reply processor/src/validators/payment.validators.ts]: I fixed this

**Review by Tung-Huynh-Shopmacher — APPROVED** [review]

**Pull request merged**